File: baekjoon/Level/Silver/2644. 촌수계산/촌수계산.swift
let n = Int(readLine()!)!
let target = readLine()!.split(separator: " ").map{ Int($0)! }
let (t1, t2) = (target[0], target[1])
let m = Int(readLine()!)!

var graph: [[Int]] = Array(repeating: [], count: n + 1)

for _ in 0..<m {
    let input = readLine()!.split(separator: " ").map{ Int($0)! }
    let (u, v) = (input[0], input[1])
    graph[u].append(v)
    graph[v].append(u)
}

var visited: [Bool] = Array(repeating: false, count: n + 1)

func bfs(_ v: Int) -> Int {
    var queue = [(v, 0)]
    var idx = 0
    visited[v] = true
    
    while idx < queue.count {
        let n = queue[idx].0
        let depth = queue[idx].1
        
        if n == t2 {
            return depth
        }
        
        for i in graph[n] {
            if !visited[i] {
                queue.append((i, depth + 1))
                visited[i] = true
            }
        }
        idx += 1
    }
    return -1
}

print(bfs(t1))

File: baekjoon/Level/Silver/2644. 촌수계산/촌수계산.py
from collections import deque

n = int(input())
t1, t2 = map(int, input().split(" "))
m = int(input())

graph = [[] for i in range(n + 1)]

for _ in range(m):
    u, v = map(int, input().split(" "))
    graph[u].append(v)
    graph[v].append(u)

visited = [False] * (n + 1)

def bfs(v):
    queue = deque([(v, 0)])
    visited[v] = True

    while len(queue) > 0:
        n = queue.popleft()
        node = n[0]
        depth = n[1]

        if node == t2:
            return depth

        for i in graph[node]:
            if not visited[i]:
                queue.append((i, depth + 1))
                visited[i] = True
    return -1

print(bfs(t1))